tools/ci_build/build.py

```diff
@@ -311,6 +311,12 @@ def Invalid_Hetero_Build():
         help="Build with OpenVINO for specific hardware.")
     parser.add_argument(
         "--use_nnapi", action='store_true', help="Build with NNAPI support.")
+    parser.add_argument(
+        "--nnapi_minimal_api", type=int,
+        help="Minimal Android API level to enable NNAPI, should be no less thant 27")
+    parser.add_argument(
+        "--nnapi_host_api", type=int,
+        help="Host Android API level to use with NNAPI model converter")
     parser.add_argument(
```

Comment:
How does this relate to the Android build arg of --android_api? The help maybe needs expanding as it's not clear to me how to use it, and I know most of the internals.

Is this the value that we will get from the SDK if building on Android, or use the provided value when building on another platform?

Reply:
--android_api is the min targeted Android API level, for example --android_api 21, ORT will run on Android 21+, this one is the fake host api we want to pass to the EP while converting the models and tell the EP we are going to run the EP on this level or lower, so EP can get the largest subgraph when run GetCapability

If we are actually running on Android, this value will be obtained as a runtime system value, the build option is only to specify this value for running EP non-Android platform

I am not sure if I want to expose this in the build.py since we already hardcoded it, maybe leave this in the .cmake file and use --cmake_extra_defines just in case

Reply:
Should we say that the default is the latest NNAPI API version we know of, and that this is an optional parameter? 

Also, what should happen if someone specifies an API version later than our default value? Currently I think it's silently ignored. Do we need a warning if that happens or is it not worth the effort?

---
In reply to: [522709643](https://github.com/microsoft/onnxruntime/pull/5779#discussion_r522709643) [](ancestors = 522709643)

Reply:
Removed this from build.py


---
In reply to: [522883186](https://github.com/microsoft/onnxruntime/pull/5779#discussion_r522883186) [](ancestors = 522883186,522709643)